src/ba.c

```diff
@@ -1,11 +1,41 @@
 #include "gsa.h"
+#include <stdio.h>
+#include <string.h>
+#include <assert.h>
 
 void border_array(size_t n, const char x[n], int ba[n])
 {
     // FIXME
+    int c = 0, i = 1;
+    int len = strlen(x);
+    ba[0] = 0;
+    while (i < len) {
+        if (x[c] == x[i]) {
+            c++;
+            ba[i] = c;
+            i++;
+        }
+        else {
+            if(c > 0) {
+                c = ba[(c-1)];
+            } 
+            else {
+                ba[i] = 0;
+                i++;
+            }
+        }
+    }
 }
 
 void strict_border_array(size_t n, const char x[n], int ba[n])
 {
     // FIXME
-}
+
+    // lazy solution
+    border_array(n, x, ba);
+
+    for (int i = 0 ; i < n ; i++) {
+        if(ba[i-1] == ba[i]-1) ba[i] = 0;
+    }
```

Comment:
Unfortunately, this does not work. You cannot simply set the non-strict borders to zero. You have to find the longest strict border, and it is not true that the longest strict border is either the longest border or zero. Consider the string `abaabaa` and the second-to-last indeks. The string `abaaba` has the longest border `aba`, but that border is followed by `a` both in `aba[a]` and `ababa[a]` so it isn't strict. But the longest strict border is not empty either. The border `a` is strict; it is followed by `b`: `a[b]` and with `a`: `-----a[a]`. The longest strict border has length 1, not 0.